ClientTests/TabManagerTests.swift

```diff
@@ -28,56 +37,74 @@ class TabManagerTests: XCTestCase {
         super.tearDown()
     }
 
-    func testTabManagerStoresChangesInDB() {
+    func testTabManagerStateDelegateImplStoresChangesInDB() {
         let profile = TabManagerMockProfile()
+        let appDelegate = UIApplication.sharedApplication().delegate as? AppDelegate
+        XCTAssertNotNil(appDelegate)
+        appDelegate?.profile = profile
         let manager = TabManager(prefs: profile.prefs, imageStore: nil)
```

Comment:
Why? `appDelegate` is already the `stateDelegate`

What I've done with this test is remove the integration tests aspect entirely. So this test unit tests the implementation of the delegate method we have in `AppDelegate.swift`. It tests that it correctly calls the save method for tabs on the DB, which has its own unit test to ensure that tabs actually save. The subsequent tests test whether or not `storeChanges` calls whatever `stateDelegate` is set. Between the 3 the test the entire use case.